datmo/cli/command/task.py

```diff
@@ -71,15 +71,14 @@ def run(self, **kwargs):
         self.cli_helper.echo(__("info", "cli.task.run"))
 
         # Create input dictionaries
-        snapshot_dict = {
-            "environment_definition_filepath":
+        snapshot_dict = {}
+        if kwargs['environment_definition_filepath']:
+            snapshot_dict["environment_definition_filepath"] =\
                 kwargs['environment_definition_filepath']
-        }
-
         if not isinstance(kwargs['cmd'], list):
             if platform.system() == "Windows":
                 kwargs['cmd'] = kwargs['cmd']
-            else:
+            elif isinstance(kwargs['cmd'], str):
                 kwargs['cmd'] = shlex.split(kwargs['cmd'])
```

Comment:
@shabazpatel this works differently in travis than in mac so the tests are failing when this is a string and it is being split. can you look into this? I have a test for specifically passing in a string to the task command to test this and that is the one that is not working. you might be able to replicate by trying this on your ubuntu instance